ssh/ssh.go

```diff
@@ -0,0 +1,61 @@
+package ssh
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"golang.org/x/crypto/ssh"
+	"log"
+	"strings"
+
+	"github.com/SUSE/scf-secret-generator/model"
+)
+
+type SSHKey struct {
+	PrivateKey  string // Name to associate with private key
+	Fingerprint string // Name to associate with fingerprint
+}
+
+func GenerateSSHKey(secretData map[string][]byte, key SSHKey) {
+	// generate private key
+	private, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	privateBlock := pem.Block{
+		Type:    "RSA PRIVATE KEY",
+		Headers: nil,
+		Bytes:   x509.MarshalPKCS1PrivateKey(private),
+	}
+
+	// PEM encode private key
+	secretKey := strings.Replace(strings.ToLower(key.PrivateKey), "_", "-", -1)
+	secretData[secretKey] = pem.EncodeToMemory(&privateBlock)
+
+	// generate MD5 fingerprint
+	public, err := ssh.NewPublicKey(&private.PublicKey)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fingerprintKey := strings.Replace(strings.ToLower(key.Fingerprint), "_", "-", -1)
+	secretData[fingerprintKey] = []byte(ssh.FingerprintLegacyMD5(public))
+}
+
+func ParseSSHKey(keys map[string]SSHKey, configVar *model.ConfigurationVariable) {
```

Comment:
This name feels wrong; it is not parsing an SSH key; it is recording a key generation request.

Can't think of a good name right now unfortunately. `RecordSSHKeyId` looks ugly too, but would come closer to being descriptive.

Reply:
What about `RecordSSHKeyInfo` since it's not only recording the ID, but the names for the secret entries as well?

Reply:
Fixed in 9016638